### PR TITLE
Add required-packages field to include code-generator

### PR DESCRIPTION
**Pull request by sttts**
Fixes https://github.com/kubernetes/sample-controller/issues/6.

### CODE REVIEW

File: artifacts/scripts/construct.sh [hunk 1]
```diff
@@ -37,8 +37,8 @@ set -o nounset
 set -o pipefail
 set -o xtrace
 
-if [ ! $# -eq 9 ]; then
-    echo "usage: $0 repo src_branch dst_branch dependent_k8s.io_repos kubernetes_remote subdirectory source_repo_org source_repo_name is_library"
+if [ ! $# -eq 10 ]; then
+    echo "usage: $0 repo src_branch dst_branch dependent_k8s.io_repos required_k8s.io_packages kubernetes_remote subdirectory source_repo_org source_repo_name is_library"
```
Comment (mfojtik) [inline artifacts/scripts/construct.sh]: is this k8s specific?

**Comment by mfojtik** [comment]
LGTM

File: artifacts/scripts/construct.sh (hunk 1 above)
Comment (sttts) [reply artifacts/scripts/construct.sh]: no, not at all actually. Bad variable name. Will fix.

Comment (sttts) [reply artifacts/scripts/construct.sh]: Fixed

**Comment by kargakis** [comment]
cool

**Comment by sttts** [comment]
For reference, this won't be as easy with dep: https://github.com/golang/dep/issues/1696

We can only copy in the files without caring about dep being unhappy about the mutated vendor/ dir.

**Pull request merged**